### DISCUSSION TITLE
Regression with confidence interval band

### DISCUSSION THREAD

**Discussion by Tokazama** [JuliaPlots/Makie.jl#4170] [Ideas]
Here's my messy and unoptimized solution to a linear regression with a confidence band.
```julia
function regplot!(sc, x, y; nbins=min(length(x), 100), linecolor=:red, bandcolor=:pink)
    σ = std(x)
    x̄ = mean(x)
    sse = sum((x .- x̄).^2)
    over_n = 1/length(x)

    itc, slp = hcat(fill!(similar(x), 1), x) \ y
    x1, x2 = AbstractPlotting.extrema_nan(x)
    y1 = slp * x1 + itc
    y2 = slp * x2 + itc

    p1 = Point2f0(x1, y1)
    p2 = Point2f0(x2, y2)

    xmin, xmax = extrema(x)
    xrange = range(xmin, stop=xmax, length=nbins)
    err = σ .* sqrt.(over_n .+  (xrange .- x̄).^2 ./ sse)
    ŷ = xrange .* slp .+ itc

    linesegments!(sc, [p1 => p2], color=linecolor, linewidth=2)
    band!(sc, xrange, ŷ .- err, ŷ .+ err, color=bandcolor, smooth=true, transparency=true)
end
```
Here's an example of two of these overlayed
![image](https://user-images.githubusercontent.com/13888485/72985990-31acbb00-3db5-11ea-9a14-2a3eee3b8a1d.png)

Although it's pretty easy for people to do this on their own I think it's standard enough that it should be built in somehow. I doubt my method name or strategy is perfect though.

**Comment by mkborregaard** [comment]
I don't personally think it should be built in, but rather be a recipe on any model object (GLM, LsqFit etc)
